homeassistant/components/light/__init__.py

```diff
@@ -305,17 +306,22 @@ class SetIntentHandler(intent.IntentHandler):
                 preprocess_turn_on_alternatives(pars)
                 turn_light_off, off_pars = preprocess_turn_off(pars)
             if turn_light_off:
-                await light.async_turn_off(**off_pars)
+                task = light.async_request_call(light.async_turn_off, off_pars)
             else:
-                await light.async_turn_on(**pars)
+                task = light.async_request_call(light.async_turn_on, pars)
 
-            if not light.should_poll:
-                continue
+            change_tasks.append(task)
 
-            update_tasks.append(light.async_update_ha_state(True))
+            if light.should_poll:
+                poll_lights.append(light)
 
-        if update_tasks:
-            await asyncio.wait(update_tasks)
+        if change_tasks:
+            await asyncio.wait(change_tasks)
```

Comment:
We should use `gather`, so exceptions propagate correctly.

Reply:
i keep it on gather, until the safe wrapper is available